### PR TITLE
New: parseForESLint method and parser services

### PR DESCRIPTION
**Pull request by JamesHenry**
Implements proposed API in https://github.com/eslint/eslint/pull/6975

### CODE REVIEW

File: lib/ast-converter.js [hunk 1]
```diff
@@ -398,6 +398,12 @@ module.exports = function(ast, extra) {
     }
 
     /**
+     * Create a map between converted ESTreeNodes and their original TSNodes.
+     * This will be exported as part of the parser services for use in ESLint.
+     */
+    var nodeMap = new Map();
```
Comment (nzakas) [inline lib/ast-converter.js]: Come to think of it, we probably want a WeakMap here.

File: parser.js [hunk 2]
```diff
@@ -215,9 +213,44 @@ function parse(code, options) {
...
+ * @returns {object}         object containing the ESTree AST and parser services
+ */
+function parseForESLint(code, options) {
+    var typescriptAST = generateTypeScriptASTAndConfigureExtra(code, options);
+    var converted = astConverter(typescriptAST, extra);
+    return {
+        ast: converted.ast,
+        services: {
+            getOriginalTSNode: function getOriginalTSNode(node) {
+                return converted.nodeMap.get(node);
```
Comment (nzakas) [inline parser.js]: We can probably shorten to `getTSNode`.

**Comment by JamesHenry** [comment]
Now that eslint/eslint#6975 is pretty much approved for merging, I am going to update this PR with the various bits of feedback around implementation

**Comment by JamesHenry** [comment]
Closing in favour of https://github.com/eslint/typescript-eslint-parser/pull/94 where the various bit of feedback have been implemented

**Pull request closed without merging**